pypeit/core/wavecal/autoid.py

```diff
@@ -524,41 +515,115 @@ def run(self):
                 obad_slits = bad_slits.copy()
                 cntr += 1
                 if cntr > 100:
+                    msgs.warn("Breaking while loop before convergence. Check the wavelength solution!")
                     break
 
         # With the updates to the fits of each slit, determine the final fit, and save the QA
+        self.finalize_fit()
+
+        # Print the final report of all lines
+        self.report_final()
+        return
+
+    def run_kdtree(self, polygon=4, detsrch=7, lstsrch=10, pixtol=5):
```

Comment:
Please add a brief doc string on the input parameters.

Reply:
Done!